resources/js/components/utils/InitializeTanstackQuery.tsx

```diff
@@ -64,10 +79,12 @@ export const InitializeTanstackQuery: ParentComponent = (props) => {
         defaultOptions: {
           queries: {
             refetchOnReconnect: true,
-            refetchOnMount: false,
-            refetchOnWindowFocus: false,
+            // When opening a page, reload data if it's older than a couple of seconds.
+            staleTime: 5 * 1000,
             retry: false,
-            retryOnMount: false,
+            // This is very important. The default reconcile algorithm somehow breaks the data and
```

Comment:
Chyba wdepnęliśmy w jakiś nieprzygotowany feature: https://github.com/TanStack/query/pull/6125 :|

Podlinkuj może to oraz https://github.com/TanStack/query/pull/5287 żeby było więcej kontekstu.